test cases/failing/128 subproject object as a dependency/test.json

```diff
@@ -1,7 +1,7 @@
 {
   "stdout": [
     {
-      "line": "test cases/failing/128 subproject object as a dependency/meson.build:3:0: ERROR: Tried to use subproject object as a dependency."
+      "line": "test cases/failing/128 subproject object as a dependency/meson.build:3:0: ERROR: executable keyword argument 'dependencies' was of type array[SubprojectHolder] but should have been array[Dependency | InternalDependency]"
```

Comment:
I guess? Since using `subproject` seems to be largely discouraged today I thought it wasn't that useful to have a specific error message (since the current state of the art is to to use meson.override_dependency() and meson.override_find_program()), but I can add the error message if you prefer

Reply:
This actually turns out to be really hard, because a SubprojectHolder doesn't get unholdered (it's actually not a distinct type, it's an InterpreterObject with an Interpreter in it), and is defined inside Interpreter.py, which imports type_checking,py, which means that we can't have SubprojectHolder in the global namespace of type_checking.py.